### PR TITLE
Ensure that failed fetches from the Pro API result in a non zero exit code

### LINKED ISSUE #1055: Newman run should not return success with a not found collection
**I believe this bug can have huge consequence with an continuous integration system!!!**

 1. Newman Version (can be found via `newman -v`): 3.6.0
 2. OS details (type, version, and architecture): docker & osx
 3. Are you using Newman as a library, or via the CLI? cli
 3. Did you encounter this recently, or has this bug always been there:
 4. Expected behaviour:
 5. Command / script used to run Newman:
`newman run https://www.getpostman.com/collections/651de8da5b5f-not-found`
 6. Sample collection, and auxilliary files (minus the sensitive details):
 7. Screenshots (if applicable):
screen shot 2017-06-03 at 10 27 38

Steps to reproduce the problem:

`newman run https://www.getpostman.com/collections/651de8da5b5f-not-found`
`echo $?` shows `0`

### PR DESCRIPTION
**Pull request by kunagpal**
Fixes #1055

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -103,6 +104,19 @@ util = {
                     return callback(_.set(e, 'help', `the url "${location}" did not provide valid JSON data`));
                 }
 
+                var error,
+                    resource;
+
+                if ((response.statusCode !== 200) || (body && body.error)) {
```
Comment (shamasis) [inline lib/util.js]: only stop on non 200 code but, if body.error is present, use it only to extract message.

File: lib/util.js [hunk 2]
```diff
@@ -103,6 +104,19 @@ util = {
...
+                    error = new Error(_.get(body, 'error.message'));
```
Comment (shamasis) [inline lib/util.js]: If body was not containing `error`, then error will be undefined.

Comment (kunagpal) [inline lib/util.js]: Addressed

File: lib/util.js (hunk 1 above)
Comment (kunagpal) [inline lib/util.js]: Addressed

**Pull request merged**